### PR TITLE
Add core IMC non zero event test cases

### PR DESCRIPTION
**Pull request by nasastry**
Adding test cases for core IMC events, which shouldn't return zero values

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
job.log

```
JOB ID     : 4118c16f62e6af823ec8250cfa818de54f5dd96c
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-08-24T15.08-4118c16/job.log
 (1/2) perf_core_imc_non_zero_event.py:PerfCoreIMCNonZeroEvents.test_perf_cpm_cyc:  PASS (10.84 s)
 (2/2) perf_core_imc_non_zero_event.py:PerfCoreIMCNonZeroEvents.test_perf_cpm_32mhz_cyc:  PASS (10.82 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-08-24T15.08-4118c16/results.html
JOB TIME   : 23.80 s
```

File: perf/perf_core_imc_non_zero_event.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                if int(line.strip().split()[1].replace(',', '')) == 0:
+                    fail_count = fail_count + 1
+        if fail_count > 1:
+            self.fail("%s : command failed with zero count" % cmd)
+
+    @skipUnless(IS_POWER_NV, "This test is for PowerNV")
+    def test_perf_cpm_cyc(self):
+        self.parse_op('perf stat -e core_imc/CPM_CCYC/ -C 0 -I 1000 sleep 5')
+
+    @skipUnless(IS_POWER_NV, "This test is for PowerNV")
```
Comment (PraveenPenguin) [inline perf/perf_core_imc_non_zero_event.py:68]: as seems me this test has two tests and both are not supported in Power VM so I feel we can skip test case in setup it self

**Comment by nasastry** [comment]
@PraveenPenguin Done with the change.

```
JOB ID     : ddda6ee4447afb6334193d7bd0c71442bd85c19a
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-08-25T14.39-ddda6ee/job.log
 (1/2) perf_core_imc_non_zero_event.py:PerfCoreIMCNonZeroEvents.test_perf_cpm_cyc: PASS (13.34 s)
 (2/2) perf_core_imc_non_zero_event.py:PerfCoreIMCNonZeroEvents.test_perf_cpm_32mhz_cyc: PASS (10.87 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-08-25T14.39-ddda6ee/results.html
JOB TIME   : 27.34 s
```


```
JOB ID     : 999b2f82209bcf1f9f8c0fa2958fb0b7949cfed6
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-08-25T05.10-999b2f8/job.log
 (1/2) perf_core_imc_non_zero_event.py:PerfCoreIMCNonZeroEvents.test_perf_cpm_cyc: SKIP: This test is for PowerNV
 (2/2) perf_core_imc_non_zero_event.py:PerfCoreIMCNonZeroEvents.test_perf_cpm_32mhz_cyc: SKIP: This test is for PowerNV
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 2 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-08-25T05.10-999b2f8/results.html
JOB TIME   : 1.04 s
```

File: perf/perf_core_imc_non_zero_event.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        for pkg in pkgs:
+            if not smg.check_installed(pkg) and not smg.install(pkg):
+                self.cancel(
+                    "Package %s is missing/could not be installed" % pkg)
+        process.run("ppc64_cpu --frequency -t 10 &", shell=True,
+                    ignore_status=True, verbose=True, ignore_bg_processes=True)
+
+    def parse_op(self, cmd):
+        fail_count = 0
+        output = process.system_output(cmd, ignore_status=True, allow_output_check='combined', verbose=True, shell=True, sudo=True)
```
Comment (PraveenPenguin) [inline perf/perf_core_imc_non_zero_event.py:57]: @nasastry looks good to me just one style issue can this line wrap in 80 char

**Comment by nasastry** [comment]
@PraveenPenguin styling taken care.
```
JOB ID     : 9900e490064a6d4683c3b32b2cb6dcf292cdf74e
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-08-25T17.23-9900e49/job.log
 (1/2) perf_core_imc_non_zero_event.py:PerfCoreIMCNonZeroEvents.test_perf_cpm_cyc: PASS (10.88 s)
 (2/2) perf_core_imc_non_zero_event.py:PerfCoreIMCNonZeroEvents.test_perf_cpm_32mhz_cyc: PASS (10.88 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-08-25T17.23-9900e49/results.html
JOB TIME   : 24.28 s
```

**Review by PraveenPenguin — APPROVED** [review]
LGTM
